### ISSUE TITLE
Add Window Change event

### ISSUE THREAD

**Issue by astronouth7303** [xonsh/xonsh#2545]
Add an event when the terminal changes size.

On POSIX, this event is fired on `SIGWINCH`. Unfortunately, the way `PopenThread` is written makes this change non-trivial.

Research needs to be done for Win32.

(This came out of #2543.)

**Comment by astronouth7303** [comment]
(This, of course, would be easier under slug -_-)

**Comment by scopatz** [comment]
Thanks for opening this @astronouth7303. PopenThread should be compatible with any other signal handler we want to register. PopenThread.wait() replaces the existing window change handler after the thread closes.

**Comment by astronouth7303** [comment]
Right, but the coordination is complicated because the event must still be fired, even if it is delayed.
